main.tf

```
@@ -6,8 +6,9 @@ locals {
   versioning_enabled            = local.enabled && var.versioning_enabled
   transfer_acceleration_enabled = local.enabled && var.transfer_acceleration_enabled
 
-  bucket_name = var.bucket_name != null && var.bucket_name != "" ? var.bucket_name : module.this.id
-  bucket_arn  = "arn:${local.partition}:s3:::${join("", aws_s3_bucket.default.*.id)}"
+  bucket_name                      = var.bucket_name != null && var.bucket_name != "" ? var.bucket_name : module.this.id
+  bucket_arn                       = "arn:${local.partition}:s3:::${join("", aws_s3_bucket.default.*.id)}"
+  cloudtrail_bucket_policy_enabled = var.cloudtrail_bucket_policy_enabled
```

Why use a `local` if it's the same as the `var` ? I think this line should be removed and the lines below should point directly to the `var`.

```suggestion
  bucket_name = var.bucket_name != null && var.bucket_name != "" ? var.bucket_name : module.this.id
  bucket_arn  = "arn:${local.partition}:s3:::${join("", aws_s3_bucket.default.*.id)}"
```